src/Razor/src/Microsoft.AspNetCore.Razor.Language/Legacy/CSharpCodeParser.cs

```diff
@@ -2067,15 +2070,24 @@ private void ParseUsingDeclaration(in SyntaxListBuilder<RazorSyntaxNode> builder
                 else if (At(CSharpKeyword.Static))
                 {
                     // static using
+                    isStatic = true;
                     AcceptAndMoveNext();
                     AcceptWhile(IsSpacingToken(includeNewLines: false, includeComments: true));
+                    nonNamespaceTokenCount = TokenBuilder.Count;
                     TryParseNamespaceOrTypeName(directiveBuilder);
                 }
 
+                var usingStatementTokens = TokenBuilder.ToList().Nodes;
+                var usingContentTokens = usingStatementTokens.Skip(1);
+                var parsedNamespaceTokens = usingStatementTokens
+                    .Skip(nonNamespaceTokenCount)
+                    .Where(s => s.Kind != SyntaxKind.CSharpComment && s.Kind != SyntaxKind.Whitespace && s.Kind != SyntaxKind.NewLine);
```

Comment:
Just double checking we weren't completely crazy: We don't allow Razor comments in a using statement do we?

Reply:
Yup not that crazy 😄